src/main/java/tutorlink/command/UpdateComponentCommand.java

```diff
@@ -0,0 +1,77 @@
+package tutorlink.command;
+
+import tutorlink.appstate.AppState;
+import tutorlink.commons.Commons;
+import tutorlink.component.Component;
+import tutorlink.exceptions.ComponentNotFoundException;
+import tutorlink.exceptions.IllegalValueException;
+import tutorlink.exceptions.InvalidWeightingException;
+import tutorlink.exceptions.TutorLinkException;
+import tutorlink.lists.ComponentList;
+import tutorlink.result.CommandResult;
+
+import java.util.HashMap;
+
+public class UpdateComponentCommand extends Command {
+    public static final String[] ARGUMENT_PREFIXES = {"c/", "w/", "m/"};
+    public static final String COMMAND_WORD = "update_component";
+
+    @Override
+    public CommandResult execute(AppState appState, HashMap<String, String> hashmap) throws TutorLinkException {
+        String name = hashmap.get(ARGUMENT_PREFIXES[0]);
+        String cWeight = hashmap.get(ARGUMENT_PREFIXES[1]);
+        String cMark = hashmap.get(ARGUMENT_PREFIXES[2]);
+        if (name == null || (cWeight == null && cMark == null)) {
+            throw new IllegalValueException("Error! n/ is required, and one of w/ or m/ is required");
+        }
+        Integer weight = null;
+        Double mark = null;
+        try {
+            if (cWeight != null) {
+                weight = Integer.parseInt(cWeight);
+            }
+            if (cMark != null) {
+                mark = Double.parseDouble(cMark);
+            }
+        } catch (NumberFormatException e) {
+            throw new IllegalValueException("Error! weight is not an integer or mark is not a real positive number");
+        }
```

Comment:
You should split the two cases so that users will know which parameter is incorrect.

Reply:
Can be more descriptive in your errors. Some reviewers have commented that the error messages are not descriptive enough.